TCAT/Controllers/RouteDetail+ContentViewController.swift

```diff
@@ -227,97 +221,129 @@ class RouteDetailContentViewController: UIViewController, GMSMapViewDelegate, CL
+                                 }
+                                 
+                                 if self.noDataRouteList.isEmpty {
+                                    self.hideBanner()
+                                 }
+                                 
+                                 self.setBusLocation(busLocation)
+                                
+                            } // switch end
+                            
+                        } else {
+                            print("Error: Successful response, but no content. Likely client-side issue")
+                        }
+                        
+                }) { (error) in
+                    
+                    print("RouteDetailVC getBusLocations Error:", error)
+                    self.showBanner("Cannot connect to live tracking", status: .danger)
+                    
+                } // network completion handler end
                 
-            }
+            } // depart conditional end
+            
+        } // for loop end
 
-        }
+    }
+
+    /** Update the map with new busLocations, adding or replacing based on vehicleID.
+        If `validTripIDs` is passed in, only buses that match the tripID will be drawn.
+        The input includes every bus associated with a certain line.
+     */
+    func setBusLocation(_ bus: BusLocation) {
+        
+        let busCoords = CLLocationCoordinate2DMake(bus.latitude, bus.longitude)
+        let existingBus = buses.first(where: {
+            return ($0.userData as? BusLocation)?.vehicleID == bus.vehicleID
+        })
         
-        // if a bus on the map doesn't appear in latest buses, remove it
-        for (index, mappedBus) in buses.enumerated() {
+        // If bus is already on map, update and animate change
+        if let newBus = existingBus {
             
-            if !busLocations.contains(where: { (newBusLocation) -> Bool in
-                return (mappedBus.userData as? BusLocation)?.vehicleID == newBusLocation.vehicleID
-            }) {
-                mappedBus.map = nil
-                buses.remove(at: index)
-            }
+            /// Allow time to receive new live bus request
+            let latencyConstant = 0.25
+            
+            CATransaction.begin()
+            CATransaction.setAnimationDuration(networkRefreshRate + latencyConstant)
+            newBus.userData = bus
+            (newBus.iconView as? BusLocationView)?.setBearing(bus.heading, start: existingBus!.position, end: busCoords)
+            newBus.position = busCoords
+            CATransaction.commit()
             
         }
-
+            
+        // Otherwise, add bus to map
+        else {
+            
+            let marker = GMSMarker(position: busCoords)
+            (bus.iconView as? BusLocationView)?.setBearing(bus.heading)
+            marker.iconView = bus.iconView
+            // marker.appearAnimation = .pop
+            // marker.tracksViewChanges = true
```

Comment:
do u need the commented out code? ow i would delete it

Reply:
Figured out a solution!